ruby/true_automation/lib/true_automation/driver/appium_lib.rb

```diff
@@ -0,0 +1,45 @@
+require_relative '../client'
+
+module TrueAutomation
+  module Driver
+    class AppiumLib < Appium::Core::Driver
+      def initialize(opts = {})
+        @delegate_target = self # for testing purpose
+        @automation_name = nil # initialise before 'set_automation_name'
+
+        opts = Appium.symbolize_keys opts
+        validate_keys(opts)
+
+        @custom_url = opts.delete :url
+        @caps = get_caps(opts)
+
+        set_appium_lib_specific_values(get_appium_lib_opts(opts))
+        set_app_path
+        set_appium_device
+        set_automation_name
+
+        extend_for(device: @device, automation_name: @automation_name)
+
+        @ta_client = TrueAutomation::Client.new
```

Comment:
because it initialize

Reply:
I mean if it is only change, you can use it in method which is needed or even define getter method like:
```
def ta_client
  @_ta_client ||= TrueAutomation::Client.new
end
```

Reply:
I think its not a good idea to create getter with memo, looks like helper_method not oop

Reply:
It is not an expensive to declare instance variable
```
The advantage of memoization is that it allows your objects to be lazy and not initialize their state until you need it. But sometimes you have state that you will always need to initialize, even though it’s expensive. In those cases, consider setting the state as part of the initializer and using a simple attr_reader to read the state. This is Ruby, after all; if there’s one way to do it, there are a dozen other ways as well.
```

Reply:
Ta client could be not initialized only in case it this different class without overridden initializer.
So in case of helper, it will raise an exception method_missing instead of null pointer exception

But in case of separate method you don't need to override any initializer

Reply:
But it is ok to override. Matter of taste